tests/hardhat/Shortfall.ts

```diff
@@ -614,3 +629,108 @@ describe("Shortfall: Tests", async function () {
     });
   });
 });
+
+describe("Shortfall: Deflationary token Scenario", async function () {
+  before(async () => {
+    await mine(1000);
+    await setup();
+  });
+  let startBlockNumber;
+  it("Start auction", async function () {
+    vDAI.badDebt.returns("0");
+    vWBTC.badDebt.returns("0");
+    vFloki.badDebt.returns(parseUnits("100", 18));
+    await vFloki.setVariable("badDebt", parseUnits("100", 18));
+
+    riskFundBalance = "500";
+    fakeRiskFund.poolReserves.returns(parseUnits(riskFundBalance, 18));
+
+    await shortfall.connect(owner).updateMinimumPoolBadDebt(convertToUnit(10, 18));
+
+    const receipt = await shortfall.startAuction(poolAddress);
+    startBlockNumber = receipt.blockNumber;
+
+    const auction = await shortfall.auctions(poolAddress);
+    expect(auction.status).equal(1);
+    expect(auction.auctionType).equal(1);
+  });
+
+  it("Place bid", async function () {
+    const auction = await shortfall.auctions(poolAddress);
+
+    await mockFloki.approve(shortfall.address, parseUnits("100", 18));
+
+    const previousFlokiBalance = await mockFloki.balanceOf(shortfall.address);
+    const previousBidderBalance = await mockFloki.balanceOf(bidder1.address);
+
+    await shortfall.connect(bidder1).placeBid(poolAddress, auction.startBidBps);
+
+    const afterFlokiBalance = await mockFloki.balanceOf(shortfall.address);
+    const afterBidderBalance = await mockFloki.balanceOf(bidder1.address);
+
+    expect(previousFlokiBalance).equal(convertToUnit("0", 18));
+    expect(previousBidderBalance).equal(convertToUnit("198", 18));
+
+    expect(afterFlokiBalance).equal(convertToUnit("99", 18));
+    expect(afterBidderBalance).equal(convertToUnit("98", 18));
+  });
+
+  it("Transfer back previous balance after second bid", async function () {
+    const previousFlokiBalance = await mockFloki.balanceOf(shortfall.address);
+    const previousBidderBalance = await mockFloki.balanceOf(bidder1.address);
+    const previousBidder2Balance = await mockFloki.balanceOf(bidder2.address);
+
+    await shortfall.connect(bidder2).placeBid(poolAddress, "120");
+
+    const afterFlokiBalance = await mockFloki.balanceOf(shortfall.address);
+
+    const afterBidderBalance = await mockFloki.balanceOf(bidder1.address);
+    const afterBidder2Balance = await mockFloki.balanceOf(bidder2.address);
+
+    expect(previousFlokiBalance).equal(convertToUnit("99", 18));
+    expect(previousBidderBalance).equal(convertToUnit("98", 18));
+    expect(previousBidder2Balance).equal(convertToUnit("198", 18));
+
+    expect(afterFlokiBalance).equal(convertToUnit("99", 18));
+    expect(afterBidderBalance).equal("196010000000000000000");
+    expect(afterBidder2Balance).equal(convertToUnit("98", 18));
```

Comment:
There's some cool stuff in hardhat-chai-matchers:

```
const tx = await shortfall.connect(bidder2).placeBid(poolAddress, "120");
await expect(tx).to.changeTokenBalance(mockFloki, bidder1.address, expectedBalanceDelta);
```

Reply:
Thanks @kkirka , I have updated the tests: 492f60496d8d9c38be3e377cb4816a2e6390227b